### ISSUE TITLE
Pass-throughs should not use `-map`

### ISSUE THREAD

**Issue by kkroening** [kkroening/ffmpeg-python#23]
This currently produces the wrong ffmpeg args:
```
>>> ' '.join(ffmpeg
...     .merge_outputs(
...         (ffmpeg
...             .input('in1.mp4')
...             .output('out1.mp4')
...         ),
...         (ffmpeg
...             .input('in2.mp4')
...             .output('out2.mp4')
...         )
...     )
...     .get_args()
... )
u'-i in1.mp4 -i in2.mp4 out1.mp4 -map [1] out2.mp4'
```

The `-map` should not be there, since no `-filter_complex` param is needed.

Should be fairly easy to fix by omitting `-map` for passthroughs and generating output params in the correct order.

See `test_multi_passthrough` in `feature/17` (or `master` once #17 is merged).

**Comment by kkroening** [comment]
Hmm, actually it might be a little more non-trivial. Take the following for example:

```
ffmpeg.view([
        (ffmpeg
            .input('in1.mp4')
            .output('out1.mp4')
        ),
        (ffmpeg
            .input('in2.mp4')
            .output('out2.mp4')
        ),
        (ffmpeg
            .input('in3.mp4')
            .hflip()
            .output('out3.mp4')
        ),
        (ffmpeg
            .input('in3.mp4')
            .vflip()
            .output('out4.mp4')
        ),
        (ffmpeg
            .input('in5.mp4')
            .hflip()
            .split()[0]
            .output('out5.mp4')
        ),
])
```

<img width="604" alt="screen shot 2017-07-09 at 9 21 16 pm" src="https://user-images.githubusercontent.com/2035953/28001763-9d04b96a-64ec-11e7-85a9-ec23c70266cd.png">

This has four connected components, two of which are passthroughs and two of which use filters.

 For fully connected graphs, this is a non-issue, so it seems to be an issue if one or more passthroughs is used _and_ multiple connected components are present.

There are two solutions that I know of:
- Inject a dummy filter (e.g. `split=n=1`) into passthroughs so that they become part of the `-filter_complex` and thus get a stream number assigned so that `-map` can be used. Probably the simplest way to go.

- Detect the individual connected components and output each passthrough individually, and combine any that use filters. The picture above would be broken up as follows:

<img width="604" alt="screen shot 2017-07-09 at 9 21 16 pm" src="https://user-images.githubusercontent.com/2035953/28002008-5f072baa-64ee-11e7-92c5-5173b49c7cf0.png">

The two passthroughs are highlighted in orange and the two filter-based components are combined in the purple box.

This is certainly more complicated but it gets rid of the need for an extra dummy filter. I'm not sure if a dummy filter affects performance in any way, so it's almost certainly best to do it the simple way first and then optimize later. Good to know that there's a way to do it in a more ideal way if needed though.

**Comment by kkroening** [comment]
(oops, forgot a split in the `in3` -> `out3` +`out4` component; oh well...)

**Comment by kkroening** [comment]
Fixed by #103.
